app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -154,6 +156,28 @@ protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Icepick.restoreInstanceState(this, savedInstanceState);
 
+        final Intent intent = getIntent();
```

Comment:
This variable could have a more meaningful name. Even something like `newIntent` may be an improvement.

Reply:
Very useful suggestion, I will improve it.